Refactoring pass 3

orhanna14 opened the pull request:
Yess! Addressed all of your comments from the last pass, and finally figured out how to extract logic from the template, yahoo!

baccigalupi commented on lib/game_runner.rb:
```diff
@@ -11,27 +13,40 @@ def initialize(stdout, stdin, grid = Grid.new)
-  def get_user_input
-    stdin.gets.chomp
+  def mark_grid
+    grid.mark(valid_coordinate)
```
Question: Does your flow ask for the user for input twice? 

The run method says `valid_coordinate` and then `mark_grid` which calls `valid_coordinate`. That implies that `valid_coordinate` which seems to ask for user input asks for it twice?

orhanna14 replied:
Ah interesting! It does not prompt the user for input twice, but I removed it `valid_coordinate` from the run method because it wasn't necessary.

baccigalupi commented on lib/grid_printer.rb:
```diff
@@ -0,0 +1,29 @@
+   grid_template_model_mustache.gsub(/[A-C][1-3]/) { |cell| grid.get_value(cell) }
+  end
+
+  def grid_template_model_mustache
```
I wouldn't call this mustache since it isn't really mustache. It's just a grid template!

orhanna14 replied:
LOL oops 😆

orhanna14 commented:
To answer your questions: 

> There were some other comments in there about:

- Classes needing to be initialized with the grid vs passing them into methods.

- Flow around getting valid coordinates vs user input and where the final coordinate lives.

Yes, could we talk through these? Sync chat or video chat tomorrow works for me :)

> Also, I noticed there are three open PRs, but didn't check them out. Did you merge all your accepted PRs yet?

Oops, yes. Those were older versions of making these tests pass and refactoring. They're closed now. 

> I was also thinking maybe we should make these smaller and commit the working code, and then tackle each issue in a new PR. Thoughts?

I like where your head is at! That sounds good to me. I'll go ahead and merge this branch and then the next PR's will be addressing the comments above.
